src/components/ProgressBar/ProgressBar.js

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useState, useRef } from 'react';
+import styled from 'styled-components';
+import { theme } from './../styles/theme';
+import { useSpring, animated } from 'react-spring';
+
+function ProgressBar(props) {
+    const { children, pause, animationTime, reset } = props;
+    const pathRef = useRef();
+
+    const [resetAnimation, setResetAnimation] = useState(false);
+
+    const durationTime = animationTime * 1000;
+
+    const animationParameters = useSpring({
```

Comment:
hey, I don't think you want spring for this animation. You want something linear, and spring is physics simulated animation normally used for bounces.

I would suggest you simply use math here for calculating the `stroke-dashoffset`, something like `timer * 283 / animationTime` (https://en.wikipedia.org/wiki/Cross-multiplication)

Reply:
and I don't see why you would need any hook on this component, passing the computed stroke dashoffset would be enought.

I would use CSS `transition: stroke-dashoffset 1s linear` in the path and make sure that `useTimer` is ticking every second, to sync with the CSS transition.